File: 1 - General/Vishal/Vishal S/Easy/Valid Parentheses/Solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        stk = []
        for x in s:
            if x in ["(", "{", "["]:
                stk.append(x)
            else:
                if stk and ( (x == ")" and stk[-1] == "(") or (x=="}" and stk[-1] == "{") or (x=="]" and stk[-1]=="[")):
                    stk.pop()
                else:
                    return False

        return len(stk) == 0

File: 1 - General/Vishal/Vishal S/Easy/Valid Parentheses/Solution.go
func isValid(s string) bool {
    stk := []string{}
    for _, x_rune := range s{
        x := string(x_rune)
        if x == "(" || x == "{" || x =="[" {
            stk = append(stk, x)
        } else{
            l := len(stk)
            if l > 0 && ((x == ")" && stk[l-1]=="(") || (x == "]" && stk[l-1]=="[") || (x == "}" && stk[l-1]=="{")) {
                stk = stk[0:l-1]
            }else{
                return false
            }
        }
    }
    return len(stk) == 0
}